sdk/python/tests/integration/registration/test_sql_registry.py

```diff
@@ -0,0 +1,588 @@
+        and feature_view.features[2].dtype == Array(String)
+        and feature_view.features[3].name == "fs1_my_feature_4"
+        and feature_view.features[3].dtype == Array(Bytes)
+        and feature_view.entities[0] == "fs1_my_entity_1"
+    )
+
+    sql_registry.delete_feature_view("my_feature_view_1", project)
+    feature_views = sql_registry.list_feature_views(project)
+    assert len(feature_views) == 0
+
+    sql_registry.teardown()
+
+
+@pytest.mark.skipif(
+    sys.platform == "darwin", reason="does not run on mac github actions"
+)
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "sql_registry", [lazy_fixture("mysql_registry"), lazy_fixture("pg_registry")],
+)
+def test_apply_data_source(sql_registry):
+    # Create Feature Views
+    batch_source = FileSource(
+        name="test_source",
+        file_format=ParquetFormat(),
+        path="file://feast/*",
+        timestamp_field="ts_col",
+        created_timestamp_column="timestamp",
+    )
+
+    entity = Entity(name="fs1_my_entity_1", join_keys=["test"])
+
+    fv1 = FeatureView(
+        name="my_feature_view_1",
+        schema=[
+            Field(name="fs1_my_feature_1", dtype=Int64),
+            Field(name="fs1_my_feature_2", dtype=String),
+            Field(name="fs1_my_feature_3", dtype=Array(String)),
+            Field(name="fs1_my_feature_4", dtype=Array(Bytes)),
+        ],
+        entities=[entity],
+        tags={"team": "matchmaking"},
+        batch_source=batch_source,
+        ttl=timedelta(minutes=5),
+    )
+
+    project = "project"
+
+    # Register data source and feature view
+    sql_registry.apply_data_source(batch_source, project, commit=False)
+    sql_registry.apply_feature_view(fv1, project, commit=True)
+
+    registry_feature_views = sql_registry.list_feature_views(project)
+    registry_data_sources = sql_registry.list_data_sources(project)
+    assert len(registry_feature_views) == 1
+    assert len(registry_data_sources) == 1
+    registry_feature_view = registry_feature_views[0]
+    assert registry_feature_view.batch_source == batch_source
+    registry_data_source = registry_data_sources[0]
+    assert registry_data_source == batch_source
+
+    # Check that change to batch source propagates
+    batch_source.timestamp_field = "new_ts_col"
+    sql_registry.apply_data_source(batch_source, project, commit=False)
+    sql_registry.apply_feature_view(fv1, project, commit=True)
+    registry_feature_views = sql_registry.list_feature_views(project)
+    registry_data_sources = sql_registry.list_data_sources(project)
+    assert len(registry_feature_views) == 1
+    assert len(registry_data_sources) == 1
+    registry_feature_view = registry_feature_views[0]
+    assert registry_feature_view.batch_source == batch_source
+    registry_batch_source = sql_registry.list_data_sources(project)[0]
+    assert registry_batch_source == batch_source
+
+    sql_registry.teardown()
```

Comment:
mybe also an integration test that tests apply?

Reply:
Is there a reason we need an integration test equivalent? Since the tests are running against a local instance of pg/mysql?